packages/contracts-bedrock/scripts/Artifacts.s.sol

```diff
@@ -21,11 +22,134 @@ struct Deployment {
+        // Now recover the broadcasting state, if any
+        if(_callerMode == VmSafe.CallerMode.RecurrentBroadcast) {
+            vmSafe.startBroadcast(_msgSender);
+        }
+        if(_callerMode == VmSafe.CallerMode.Broadcast) {
+            vmSafe.broadcast(_msgSender);
+        }
+    }
+
+    /// @notice Returns all of the deployments done in the current context.
+    function newDeployments() external view returns (Deployment[] memory out_) {
+        (VmSafe.CallerMode callerMode, address msgSender) = _noBroadcastPre();
+        out_ = deploymentRegistry.newDeployments();
+        _noBroadcastPost(callerMode, msgSender);
+    }
+
+    /// @notice Returns whether or not a particular deployment exists.
+    /// @param _name The name of the deployment.
+    /// @return out_ Whether the deployment exists or not.
+    function has(string memory _name) public view returns (bool out_) {
+        (VmSafe.CallerMode callerMode, address msgSender) = _noBroadcastPre();
+        out_ = deploymentRegistry.has(_name);
+        _noBroadcastPost(callerMode, msgSender);
+    }
+
+    /// @notice Returns the address of a deployment. Also handles the predeploys.
+    /// @param _name The name of the deployment.
+    /// @return out_ The address of the deployment. May be `address(0)` if the deployment does not
+    ///         exist.
+    function getAddress(string memory _name) public view returns (address payable out_) {
+        (VmSafe.CallerMode callerMode, address msgSender) = _noBroadcastPre();
+        out_ = deploymentRegistry.getAddress(_name);
+        _noBroadcastPost(callerMode, msgSender);
+    }
+
+    /// @notice Returns the address of a deployment and reverts if the deployment
+    ///         does not exist.
+    /// @return out_ The address of the deployment.
+    function mustGetAddress(string memory _name) public view returns (address payable out_) {
+        (VmSafe.CallerMode callerMode, address msgSender) = _noBroadcastPre();
+        out_ = deploymentRegistry.mustGetAddress(_name);
+        _noBroadcastPost(callerMode, msgSender);
+    }
+
+    /// @notice Returns a deployment that is suitable to be used to interact with contracts.
+    /// @param _name The name of the deployment.
+    /// @return out_ The deployment.
+    function get(string memory _name) public view returns (Deployment memory out_) {
+        (VmSafe.CallerMode callerMode, address msgSender) = _noBroadcastPre();
+        out_ = deploymentRegistry.get(_name);
+        _noBroadcastPost(callerMode, msgSender);
+    }
+
+    /// @notice Appends a deployment to disk as a JSON deploy artifact.
+    /// @param _name The name of the deployment.
+    /// @param _deployed The address of the deployment.
+    function save(string memory _name, address _deployed) public {
+        (VmSafe.CallerMode callerMode, address msgSender) = _noBroadcastPre();
+        deploymentRegistry.save(_name, _deployed);
```

Comment:
This method is state changing, so in this case you would need to keep the current wrappers